tcl/org.testeditor.tcl.dsl.ide/src/org/testeditor/tcl/dsl/ide/TclContentProposalProvider.xtend

```diff
@@ -100,35 +132,55 @@ class TclContentProposalProvider extends IdeContentProposalProvider {
 		}
 	}
 
-	private def boolean isLocatedWithinMacroUsage(ContentAssistContext context) {
-		return (context.currentModel instanceof MacroTestStepContext && !(context.previousModel instanceof TestStep))
-	}
-
-	private def boolean isLocatedWithinComponentUsage(ContentAssistContext context) {
-		return (context.currentModel instanceof ComponentTestStepContext && !(context.previousModel instanceof TestStep))
-	}
-
-	private def boolean isOutsidePreviousTestStep(ContentAssistContext context) {
-		return !(context.previousModel instanceof StepContent && context.previousModel.isIncompleteTestStep)
-	}
-
 	private def void makeStepContentElementProposals(ContentAssistContext context, IIdeContentProposalAcceptor acceptor) {
 		val model = context.currentModel
-		if (model instanceof TestStep) {
+		if (model instanceof TestStepContext) {
+			model.envParams.forEach[makeEnvParamProposal(context, acceptor)]
+			model.enclosingMacroParameters.forEach[makeMacroParameterProposal(context, acceptor)]
+			if (context.previousModel.isIncompleteTestStep) {
+				if (model instanceof ComponentTestStepContext) {
+					model.component?.makeComponentTemplateProposal(context, acceptor)
+				}
+				if (model instanceof MacroTestStepContext) {
+					model.macroCollection?.makeMacroTemplateProposal(context, acceptor)
+				}
+			}
+		} else if (model instanceof TestStep) {
 			model.envParams.forEach[makeEnvParamProposal(context, acceptor)]
 			model.enclosingMacroParameters.forEach[makeMacroParameterProposal(context, acceptor)]
 			if (model.componentContext !== null) { // since the macro context does not allow for element parameter, yet, only component contexts are relevant here
 				model.makeElementParamsProposal('', context, acceptor)
 			}
 			if (context.previousModel.isIncompleteTestStep) {
-				model.componentContext?.component?.makeComponentTemplateProposal(model, context, acceptor)
+				model.componentContext?.component?.makeComponentTemplateProposal(context, acceptor)
 				model.macroContext?.macroCollection?.makeMacroTemplateProposal(context, acceptor)
 			}
+			makeStandardStepElementProposals(context, acceptor)
 		} else if (model instanceof StepContentElement) {
 			model.makeElementParamsProposal(context, acceptor)
 		}
 	}
 
+	private def void makeStandardStepElementProposals(ContentAssistContext context, IIdeContentProposalAcceptor acceptor) {
+		val myContext = context.copy
+		myContext.prefix = context.prefix.trim
+		val proposalStepTexts = if (context.prefix == '\n') {
+				#['"value"', "@", "?", ".", "<", "<>"]
+			} else {
+				#['"value"', "@", "?", ".", "<", "<>", "value"]
```

Comment:
ok, so after a newline, you can, among other things, continue with "value" (_with_ the quotes), otherwise the quotes are optional(?)

Would be nice if these two cases would be more descriptive.
Also, does it make sense to introduce constants for these "magic strings"? Or, even better, use existing constants from the language, but I guess Xtext does not generate something like that (e.g. from token definitions in the grammar), does it?